src/anyio/_core/_fileio.py

```diff
@@ -1,25 +1,106 @@
+import array
+import mmap
```

Comment:
I suspect mmap might not always exist

```
>>> import mmap
>>> mmap
<module 'mmap' from '/usr/lib/python3.8/lib-dynload/mmap.cpython-38-x86_64-linux-gnu.so'>
```

Reply:
use this instead of copying from _typeshed I think

```
if TYPE_CHECKING:
    from _typeshed import ReadableBuffer, WritableBuffer
else:
    ReadableBuffer = WritableBuffer = object
```

Reply:
I'm no longer importing these modules at run time.